### PR TITLE
Capybara

### PR DESCRIPTION
**Pull request by Andyveloper**
# Changelog

- The application now has proper tests
- Tests have been implemented for the different views
- The application now has eager loading working
- Performances Improvements

## Hotfixes
- Code format improved

### CODE REVIEW

File: spec/features/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    it 'shows the number of posts' do
+      visit root_path
+      expect(page).to have_content(@user.posts.count)
+    end
+
+    it 'See more button should redirect to user profile' do
+      visit root_path
+      expect(page).to have_css("a[href*='#{user_path(@user.id)}']")
+    end
```
Comment (omar-muhamad) [inline spec/features/user_index_spec.rb]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=When%20I%20click%20on%20a%20user%2C%20I%20am%20redirected%20to%20that%20user%27s%20show%20page.), Please use the correct way to test it.
[Hint] You can use `click_link` and `have_current_path` to test it right like that:
```ruby
it 'redirects to the user's show page when you click on a user' do
      visit root_path
      click_link( add the link )
      expect(page).to have_current_path( add the user show path )
    end
```
see this [capybara cheat sheet](https://gist.github.com/zhengjia/428105) to know more.

File: spec/features/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  it 'shows a button that let me view user posts' do
+    visit user_path(@user.id)
+    expect(page).to have_content('See all posts')
+  end
+
+  it 'redirects me to posts show page' do
+    visit user_path(@user.id)
+    expect(page).to have_css("a[href*='/users/#{@user.id}']")
+  end
```
Comment (omar-muhamad) [inline spec/features/user_show_spec.rb]: - [ ] Same here, As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=When%20I%20click%20a%20user%27s%20post%2C%20it%20redirects%20me%20to%20that%20post%27s%20show%20page.), Please use the correct way to test it using `click_link` and `have_current_path` as I showed above.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Andyveloper and @indigodavid,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Gitflow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please add the test database name to `config/database.yml` because it's mandatory to be exist in this file.
[Hint] See this image to know how to do it:
image
 

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/features/user_index_spec.rb (hunk 1 above)
Comment (Andyveloper) [reply spec/features/user_index_spec.rb]: Thank you very much for your comments and for your help

File: spec/features/post_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  end
+
+  it 'should have pagination button' do
+    expect(page).to have_content('Pagination')
+  end
+
+  it 'should redirect to post show' do
+    click_on @post1.id
+    expect(page).to have_current_path(user_post_path(@user, @post1))
+  end
```
Comment (Andyveloper) [inline spec/features/post_index_spec.rb:62]: Thank you, my only comment about this one specifically is that this one actually has `have_current_path` and `click_on` is literally the same as `click_link`

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @Andyveloper  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Linter checks passing ✔️ 
- Made changes as requested by the previous reviewer 👍 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**